autodist/autodist.py

```diff
@@ -97,12 +101,18 @@ def build_strategy(self):
         """
         return self._strategy_builder.build(self._original_graph_item, self._resource_spec)
 
-    def _build_or_load_strategy(self):
+    def _build_or_load_strategy(self, load=False):
         self._original_graph_item.prepare()
         if IS_AUTODIST_CHIEF:
             s = self.build_strategy()
             s.serialize()
         else:
+            # At AdaptDL mode, when the worker pass through this before
+            # the chief has created the strategy, this should returns
+            # nothing. Later, when the chief has created the strategy,
+            # it can load it.
```

Comment:
Still not quite sure about the purpose  of `load` and what this comment means. In L162-L163 `load` is always true when `IS_ADAPTDL` is true. Could you explain more?

Reply:
it is kind of subtle. Every worker will run through _build, and thus call _build_or_load_strategy. The first time the worker gets None from this function. The second time the worker will get the strategy from the chief.